contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -3432,6 +3473,241 @@ bbf_ProcessUtility(PlannedStmt *pstmt,
 				}
 				break;
 			}
+		case T_GrantStmt:
+			{
+				GrantStmt *grant = (GrantStmt *) parsetree;
+				char	   *dbname = get_cur_db_name();
+				const char *current_user = GetUserNameFromId(GetUserId(), false);
+				/* If object is table. */
+				if (sql_dialect != SQL_DIALECT_TSQL || grant->targtype != ACL_TARGET_OBJECT)
+					break;
+				Assert(list_length(grant->objects) == 1);
+				if (grant->objtype == OBJECT_SCHEMA)
+						break;
+				else if (grant->objtype == OBJECT_TABLE)
+				{
+					/* Don't add any entry for the GRANT statements executed while REVOKE on Schema. */
+					if (pstmt->stmt_len == 1 && grant->is_grant)
+					{
+						if (prev_ProcessUtility)
+							prev_ProcessUtility(pstmt, queryString, readOnlyTree, context, params,
+								queryEnv, dest, qc);
+						else
+							standard_ProcessUtility(pstmt, queryString, readOnlyTree, context, params,
+								queryEnv, dest, qc);
+						return;
+					}
+					/* Ignore CREATE database subcommands */
+					if (!(pstmt->stmt_len == 18 || pstmt->stmt_len == 19))
```

Comment:
We should compare the querystring. It can be broken if some new subcommand is added with same length.